-Controller-Service layer for authentication added

AndriiOsidach opened the pull request:
-DTO for login data added
-Exception for non-existing user added

AndriiOsidach committed:
-Controller-Service layer for authentication added
-DTO for login data added
-Exception for non-existing user added

vitalik12345678 commented on src/main/java/com/lpnu/shaggybeavers/dto/LoginDTO.java:
```diff
@@ -0,0 +1,10 @@
+
+import lombok.Data;
+
+@Data
```
DOn't use data because it has more side effects

vitalik12345678 commented on src/main/java/com/lpnu/shaggybeavers/exception/NotExistsUserException.java:
```diff
@@ -0,0 +1,7 @@
+package com.lpnu.shaggybeavers.exception;
+
+public class NotExistsUserException extends NotExistsObjectException{
```
I prefer don't use exception hierarchy, just extends from RuntimeException or IllegalExceptions

vitalik12345678 commented on src/main/java/com/lpnu/shaggybeavers/exception/NotExistsUserException.java:
```diff
@@ -0,0 +1,7 @@
+
+public class NotExistsUserException extends NotExistsObjectException{
+    public NotExistsUserException() {
+        super("User doesn't exist");
```
Add it like constant and constructor withou arg for with that constant message
And construcotr with arg when you check message on blank

vitalik12345678 commented on src/main/java/com/lpnu/shaggybeavers/service/impl/UserServiceImpl.java:
```diff
@@ -17,4 +18,10 @@ public class UserServiceImpl extends CRUDServiceImpl<User,Long> implements UserS
+    @Override
+    public User findByEmail(String email) {
+        return repository.findByEmail(email)
+                .orElseThrow(NotExistsUserException::new);
```
It is un flexible creating exception under each service, you can reuse NotExistObjectException with specif message

vitalik12345678 commented on src/main/java/com/lpnu/shaggybeavers/service/impl/UserServiceImpl.java:
```diff
@@ -17,4 +18,10 @@ public class UserServiceImpl extends CRUDServiceImpl<User,Long> implements UserS
     }
+
+    @Override
+    public User findByEmail(String email) {
```
Use Transactional(readOnly)

AndriiOsidach committed:
-@Data removed
-Custom exception removed. NonExistingObjectException reused
-@Transactional added

vitalik12345678 commented:
it had been done before your pr
